vm_phys.c

```diff
@@ -0,0 +1,241 @@
+#include <vm_phys.h>
+#include <queue.h>
+#include <malloc.h>
+#include <libkern.h>
+
+#define POW2(x) (1 << (x))
+#define PG_START(pg) ((pg)->phys_addr)
+#define PG_END(pg) ((pg)->phys_addr + (1 << (pg->order)) * PAGESIZE)
+#define PG_FREEQ(seg, i) (&(seg)->free_queues[(i)])
```

Comment:
Just a comment. I preffered older macro. I mean if you look at line:
`TAILQ_INSERT_HEAD(PG_FREEQ(seg, page->order), page, freeq);`
It's not immediately clear, what is happening here. And there is small opportunity of making a mistake, when choosing order of queue to insert page to. I'd rather prepare page, set up all necessary fields, and simply use:
`INSERT_FREE_PAGE(seg, page); `
Sure, we lose opportunity to insert at start or end, but algorithm isn't based on position of page in list, but on it's existence in list.